### ISSUE TITLE
What should the finetuning CSV look like?

### ISSUE THREAD

**Issue by janniks** [minimaxir/gpt-2-simple#159]
I get this error when trying to finetune a `.csv` file. How should the file look for this to work? Do you happen to have an example file?

```
~/.venv/lib/python3.7/site-packages/gpt_2_simple/src/load_dataset.py in load_dataset(enc, path, combine)
     37                 reader = csv.reader(fp)
     38                 for row in reader:
---> 39                     raw_text += start_token + row[0] + end_token + "\n"
     40         else:
     41             # Plain text

IndexError: list index out of range
```

Thanks for your amazing work 🎉

**Comment by janniks** [comment]
Figured it out. If anybody else has this problem. `gpt2-simple` `finetune` expects a csv file with a header line.

**Comment by anshul07683** [comment]
Figured it out. If anybody else has this problem. gpt2-simple finetune expects a csv file with a header line. not getting your point

**Comment by janniks** [comment]
`.csv` files can have an optional header (with the column names of the file). `gpt2-simple` expects the presence of a single line header. If you are getting the error `IndexError: list index out of range` while using a `.csv` file - this might be your issue. Try adding a single newline to the top of your `.csv` file.

**Comment by SeanBannister** [comment]
For anyone else trying to figure this out, I tried the blank header trick but was still getting the error, I then presumed the error was due to a limit on the line length in the file so I sorted the file by line length and uploaded the shorter lines first which worked. In the process of trying to figure out what the line length limit was I uploaded the entire file again and didn't get this error 😐 so I have no idea exactly what fixed this but apparently sorting the file by line length re-wrote the file and fixed it.

I've got more CSV fine-tunning to do so I'll report back if I figure out what the fix was.
